app/socket-apis/upsert-why.js

```diff
@@ -10,6 +15,15 @@ async function upsertWhy(pointObj, cb) {
   const userId = this.synuser.id
   pointObj.userId = userId // Add userId to the document
 
+  // Joi validation for the category
+  try {
+    await schema.validateAsync({category: pointObj.category});
+    console.log(`Category validation successful for: ${pointObj.category}`);
```

Comment:
1. In general, please remove console.log statements before final checking. 

2. I know that the example is using validateAsync but can you look into just using validate and getting rid of the try / catch / await.  It's extra overhead that's not adding value here.

Reply:
Thank you for the suggestion! I'll update the code to use **validate** instead of validateAsync and remove the try/catch and await to reduce overhead.